From a77d5aa27c48b4573e44aa96474f12f38b8c1fd6 Mon Sep 17 00:00:00 2001
From: Wilson Yang <wilson.yang@xilinx.com>
Date: Sat, 6 Jun 2009 14:53:43 -0600
Subject: [PATCH] Xilinx: LLTEMAC: mask out unused bits in BD3 and BD4

lltemac driver update for EDK 11.1 core. It is also backward compatible
to older core/driver. It is tested on ML405.

Signed-off-by: Wilson Yang <wilson.yang@xilinx.com>
Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/net/xilinx_lltemac/xlltemac_main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/xilinx_lltemac/xlltemac_main.c b/drivers/net/xilinx_lltemac/xlltemac_main.c
index 62e4676..13104be 100644
--- a/drivers/net/xilinx_lltemac/xlltemac_main.c
+++ b/drivers/net/xilinx_lltemac/xlltemac_main.c
@@ -134,10 +134,10 @@
     XLlDma_mBdWrite((BdPtr), XLLDMA_BD_USR2_OFFSET, 0)
 
 #define BdCsumGet(BdPtr) \
-    XLlDma_mBdRead((BdPtr), XLLDMA_BD_USR3_OFFSET)
+    (XLlDma_mBdRead((BdPtr), XLLDMA_BD_USR3_OFFSET) & 0xffff)
 
 #define BdGetRxLen(BdPtr) \
-    XLlDma_mBdRead((BdPtr), XLLDMA_BD_USR4_OFFSET)
+    (XLlDma_mBdRead((BdPtr), XLLDMA_BD_USR4_OFFSET) & 0x3fff)
 
 /*
  * Our private per device data.  When a net_device is allocated we will
-- 
1.7.0.4

